field/TinyLang/Field/Generator.hs

```diff
@@ -75,77 +89,53 @@ chooseUniVar = elements . uniVars
    then sets the current free unique so that the Supply monad is in a
    state where any new calls to freshUnique will generate uniques that
    we haven't already used.  You then have to call
-   adjustUniquesForVars before running the generator (see
-   boundedArbitraryExpr).
+   adjustUniquesForVars before running the generator.
 
    Presumably to do this properly we'd need to do
    all name generation in the Supply monad, but then I don't think
    we'd be able to get our Arbitrary instances to have the right types
-   (???).  Alternatively, Var could provide a function to set the
-   state of the Unique supply.
-
+   (???).
 -}
 
 -- Call this inside the exposed generators!
 adjustUniquesForVars :: MonadSupply m => Vars f -> m ()
 adjustUniquesForVars =
     supplyFromAtLeast . freeUniqueFoldable . map (forget $ _varUniq . _uniVarVar)
 
--- The variables used by our generators by default.
+-- | The variables used by our generators by default.
 defaultVars :: Vars f
 defaultVars = runSupply $ do
     let make uni name = Some . UniVar uni <$> freshVar name
     fieldVars <- traverse (make Field) ["x", "y", "z", "p", "q", "r", "s", "t"]
     boolVars  <- traverse (make Bool)  ["?a", "?b", "?c", "?d", "?e", "?f", "?g", "?h"]
     return $ fieldVars ++ boolVars
 
--- instance KnownUni f a => Arbitrary (UniVar f a) where
---     arbitrary = case knownUni @f @a of
---         Bool  -> UniVar Bool  <$> elements defaultVarsF
---         Field -> UniVar Field <$> elements defaultVarsB
-
--- | An arbitrary integer value (for use in comparisons)
-arbitraryValI :: (Field f, MonadGen m) => m (UniVal f (AField f))
-arbitraryValI = UniVal Field . fromInteger <$> arbitraryM
-
-instance (Arbitrary f, Field f) => Arbitrary (AField f) where
-    arbitrary = AField <$> arbitrary
-
-instance Arbitrary (UnOp f Bool Bool) where
-    arbitrary = return Not
-
-instance Arbitrary (UnOp f (AField f) Bool) where
-    arbitrary = return Neq0
-
-instance Arbitrary (UnOp f (AField f) (AField f)) where
-    arbitrary = elements [Neg, Inv]
+instance KnownUni f a => Arbitrary (UniVar f a) where
+    arbitrary = chooseUniVar defaultVars
 
```

Comment:
Is this really what we want?  Usually we'll have a list that includes some local variables, so are there circumstances where we definitely _don't_ want to choose one of those?

Reply:
Yes, I use this instance in the `prop_nestedELet` test. But I think you're right. Probably I just need to make a `newtype` wrapper around `UniVar`: `DefaultUniVar` and use it there.

Reply:
> Probably I just need to make a newtype wrapper around `UniVar`: `DefaultUniVar` and use it there.

Done.